src/components/forms/FormSearchField.js

```diff
@@ -1,46 +1,91 @@
 /**
- * @author psarando
+ * @author psarando sriram
  */
 import React from "react";
+import { useTranslation } from "i18n";
 
-import Autocomplete from "../../components/autocomplete/Autocomplete";
+import Autocomplete, {
+    createFilterOptions,
+} from "@material-ui/lab/Autocomplete";
 
-import getFormError from "./getFormError";
-
-import FormControl from "@material-ui/core/FormControl";
-import FormHelperText from "@material-ui/core/FormHelperText";
+import FormTextField from "./FormTextField";
 
 const FormSearchField = ({
-    field: { value, onBlur, onChange, ...field },
-    label,
-    helperText,
-    required,
-    form: { touched, errors, setFieldTouched, setFieldValue },
+    field: { value, onChange, ...field },
+    options,
+    size = "small",
+    renderCustomOption,
+    handleSearch,
+    form: { setFieldValue, ...form },
+    valueKey,
+    labelKey,
     ...props
 }) => {
-    const errorMsg = getFormError(field.name, touched, errors);
-    const onOptionSelected = (option) => {
-        setFieldValue(field.name, option ? option[props.valueKey] : "");
-    };
-    const onSearchBlur = (event) => {
-        setFieldTouched(field.name, true);
+    const [searchValue, setSearchValue] = React.useState(value);
+    const filter = createFilterOptions();
+    const { t } = useTranslation("common");
+
+    React.useEffect(() => {
+        const val = {};
+        val[valueKey] = value;
+        setSearchValue(val);
+    }, [value, setSearchValue, valueKey]);
+
+    const onOptionSelected = (event, newValue) => {
+        if (newValue) {
+            if (typeof newValue === "string") {
+                setFieldValue(field.name, newValue);
+            } else if (newValue && newValue.inputValue) {
+                // Create a new value from the user input
+                setFieldValue(field.name, newValue.inputValue);
+            } else {
+                setFieldValue(field.name, newValue[valueKey]);
+            }
+        } else {
+            setFieldValue(field.name, "");
+        }
     };
 
     return (
-        <FormControl fullWidth error={!!errorMsg}>
-            <Autocomplete
-                label={label}
-                controlShouldRenderValue
-                isClearable={!required}
-                cacheOptions
-                defaultInputValue={value}
-                onChange={onOptionSelected}
-                onBlur={onSearchBlur}
-                {...field}
-                {...props}
-            />
-            <FormHelperText>{errorMsg || helperText}</FormHelperText>
-        </FormControl>
+        <Autocomplete
+            id="searchField"
```

Comment:
One (last?) critical issue left in this PR:
In a metadata template like `Environment Ontology Terms`, all 3 ontology search fields will have this same `searchField` ID, rather than an ID like `metadataTemplate.metadata.avus[2]`.

Either this `id` should be removed so that the correct `id` gets spread into the `FormTextField` from `...props`, or else this `id` should be set to the `id` from `props`.

Reply:
I set it on the `Autocomplete` itself as recommended by the MUI.